### PR TITLE
[34] date time event duration for event

### PR DESCRIPTION
**Pull request by steve1rm**
# Pull Request template

###  What does this PR solve?

- Adds the time and date for the event duration
- Updates the state using the event
- Only uses 2 button to update both the time and date for either start or end

https://user-images.githubusercontent.com/14260802/229340829-3929e42e-c471-4011-a2ba-4582882e9a43.mp4

### CODE REVIEW

File: presentation/src/main/java/me/androidbox/presentation/event/screen/EventScreen.kt [hunk 1]
```diff
@@ -111,6 +133,26 @@ fun EventScreen(
...
+            if(isStartDateTime) {
+                eventScreenEvent(EventScreenEvent.OnStartTimeDuration(zonedDateTime))
+                eventScreenEvent(EventScreenEvent.OnStartDateDuration(zonedDateTime))
+            }
+            else {
+                eventScreenEvent(EventScreenEvent.OnEndTimeDuration(zonedDateTime))
+                eventScreenEvent(EventScreenEvent.OnEndDateDuration(zonedDateTime))
+            }
+        }
+    )
```
Comment (steve1rm) [inline presentation/src/main/java/me/androidbox/presentation/event/screen/EventScreen.kt:154]: @MKiperszmid @philipplackner In this DateTimeDialog I am setting both the time and the date for the start and end duration. So when the dialog opens the user will set the date and time in one go.

I have added a boolean flag to indicate which should be set the start or the end duration.

File: presentation/src/main/java/me/androidbox/presentation/event/screen/EventScreen.kt [hunk 2]
```diff
@@ -26,6 +35,9 @@ fun EventScreen(
     onEditDescriptionClicked: (description: String) -> Unit,
     modifier: Modifier = Modifier) {
 
+    val calendarStateTimeDate = rememberUseCaseState()
+    var isStartDateTime = false
```
Comment (philipplackner) [inline presentation/src/main/java/me/androidbox/presentation/event/screen/EventScreen.kt]: This is a side-effect (something that's not Compose related, but is inside of a composable context). Every single time `EventScreen` recomposes, this `isStartDateTime` will be false again.

File: presentation/src/main/java/me/androidbox/presentation/event/screen/EventScreen.kt [hunk 3]
```diff
@@ -78,14 +90,24 @@ fun EventScreen(
...
+                        endDate = eventScreenState.endDateDuration.formatDateTime(DATE_PATTERN),
+                        modifier = Modifier.fillMaxWidth(),
+                        onStartDurationClicked = {
+                            isStartDateTime = true
+                            calendarStateTimeDate.show()
+                        },
+                        onEndDurationClicked = {
+                            isStartDateTime = false
+                            calendarStateTimeDate.show()
+                        }
```
Comment (philipplackner) [inline presentation/src/main/java/me/androidbox/presentation/event/screen/EventScreen.kt]: I'd completely remove the `isStartDateTime` boolean for several reasons:
1. As mentioned above, it's a side-effect and will lead to undefined behavior
2. Whether your date picker dialog is shown or not (and also which one) is considered state. That boolean should therefore go in your screen state class and the ViewModel should toggle it. That way it'll also survive config changes such as toggling the device theme.

File: presentation/src/main/java/me/androidbox/presentation/event/screen/EventScreenState.kt [hunk 4]
```diff
@@ -5,12 +5,18 @@ import androidx.compose.runtime.mutableStateListOf
...
     val selectedVisitorType: VisitorType = VisitorType.ALL,
     val selectedVisitor: VisitorInfo? = null,
     val eventTitle: String = "New Event",
-    val eventDescription: String = "New Description",
+    val eventDescription: String = "Description",
     val selectedAgendaActionType: AgendaActionType = AgendaActionType.DELETE_EVENT,
+    val startTimeDuration: ZonedDateTime = ZonedDateTime.now(ZoneId.systemDefault()),
+    val endTimeDuration: ZonedDateTime = ZonedDateTime.now(ZoneId.systemDefault()),
+    val startDateDuration: ZonedDateTime = ZonedDateTime.now(ZoneId.systemDefault()),
+    val endDateDuration: ZonedDateTime = ZonedDateTime.now(ZoneId.systemDefault())
```
Comment (philipplackner) [inline presentation/src/main/java/me/androidbox/presentation/event/screen/EventScreenState.kt]: All these are not really durations, but rather a fixed time/date

**Review by philipplackner — APPROVED** [review]

File: presentation/src/main/java/me/androidbox/presentation/event/screen/EventScreen.kt (hunk 3 above)
Comment (steve1rm) [reply presentation/src/main/java/me/androidbox/presentation/event/screen/EventScreen.kt]: @philipplackner Updated and moved this to the EventScreenState and update using an event in the EventViewModel

File: presentation/src/main/java/me/androidbox/presentation/event/screen/EventScreenState.kt (hunk 4 above)
Comment (steve1rm) [reply presentation/src/main/java/me/androidbox/presentation/event/screen/EventScreenState.kt]: Updated to startTime, startDate, endTime, and endDate

**Pull request merged**